gatsby/create-node/create-preview-cards.js

```diff
@@ -1,65 +1,44 @@
-const slugify = require('slugify');
-const path = require('path');
-const { siteMetadata } = require('../../gatsby-config');
 const {
-  generateCard,
+  generateCardToBuffer,
 } = require('../util/preview-card-generator/generate-card');
+const { createFileNodeFromBuffer } = require('gatsby-source-filesystem');
 
-/**
- * Create a preview card file and put the url to the Gatsby store to be able
- * to link it in the page header.
- */
-const createPreviewMarkdown = ({ node, _, actions }) => {
-  const { createNodeField } = actions;
-  const post = node.frontmatter;
+const createPreviewCard = async (
+  title,
+  { node, _, actions, getCache, createNodeId },
+) => {
+  const { createNode, createNodeField } = actions;
 
   // we need the title in order to generate anything
-  if (!post.title) {
+  if (!title) {
     return;
   }
 
-  // we create a file name from the path (which is the page slug and more unique) but if none is available take the title
-  const fileName = `social-card---${slugify(post.path || post.title, {
-    lower: true,
-  })}.jpg`;
-  const outputFile = path.join('public', fileName);
-  const publicUrl = `${siteMetadata.siteUrl}/${fileName}`;
-
-  generateCard({ title: post.title }, outputFile).then(() => {
-    createNodeField({
-      node,
-      name: `socialCard`,
-      value: publicUrl,
-    });
+  const buffer = await generateCardToBuffer({ title });
+  const fileNode = await createFileNodeFromBuffer({
```

Comment:
finally I had time to create the file from buffer to create a proper file node instead of working around gatsby and output the file directly in static. That approach was never wrong but the 500 errors (which were caused by plain wrong urls we generated) brought me on this so. I like this much more. It's more complicated but less implicit.